docs_src/getting-started/tools/Ch-CommandLineInterface.md

```diff
@@ -1,104 +1,714 @@
 
-Remove the object (in this case the device) by name or id.
-``` bash
-edgex-cli device rm -n sample-json
-# Removed: sample-json
 ```
 
-``` bash
-edgex-cli device rm a5005e23-381a-4c69-a91d-ea299d993347
-# Removed: a5005e23-381a-4c69-a91d-ea299d993347 
+
+### `edgex-cli provisionwatcher list` 
+List provision watchers
+
+```
+$ edgex-cli provisionwatcher list
+Name         ServiceName        ProfileName  Labels  Identifiers
+TestWatcher  TestDeviceService  TestProfile  []      map[address:localhost port:1234]
+```
+
+### `edgex-cli provisionwatcher name` 
+Show information about a specific named provision watcher 
+
+```
+$ edgex-cli provisionwatcher name -n TestWatcher
+Name         ServiceName        ProfileName  Labels  Identifiers
+TestWatcher  TestDeviceService  TestProfile  []      map[address:localhost port:1234]
+```
+
+### `edgex-cli provisionwatcher rm` 
+Remove a provision watcher
+
+```
+$ edgex-cli provisionwatcher rm -n TestWatcher
+$ edgex-cli provisionwatcher list
+No provision watchers available
+```
+
+### `edgex-cli provisionwatcher update` 
+Update a provision watcher 
+
+This example gets the ID of a provision watcher, updates it using that ID and then displays  information about it to confirm that the labels were added
+
+```
+$ edgex-cli provisionwatcher add -n TestWatcher2 --identifiers "{\"address\":\"localhost\",\"port\":\"1234\"}" -p TestProfile -s TestDeviceService
+{{{v2} fb7b8bcf-8f58-477b-929e-8dac53cddc81  201} 7aadb7df-1ff1-4b3b-8986-b97e0ef53116}
+
+$ ID=`edgex-cli provisionwatcher name -n TestWatcher2 -j | jq -r '.provisionWatcher.id'`
+
+$ edgex-cli provisionwatcher update -n TestWatcher2 -i $ID --labels "label1,label2"
+{{v2} af1e70bf-4705-47f4-9046-c7b789799405  200}
+
+$ edgex-cli provisionwatcher name -n TestWatcher2
+Name          ServiceName        ProfileName  Labels           Identifiers
+TestWatcher2  TestDeviceService  TestProfile  [label1 label2]  map[address:localhost port:1234]
+
+```
+
+
+## Core-data service
+
+### `edgex-cli event add`
+Create an event with a specified number of random readings
+
+```
+$ edgex-cli event add -d Random-Integer-Device -p Random-Integer-Device -r 1 -s Int16 -t int16
+Added event 75f06078-e8da-4671-8938-ab12ebb2c244
+
+$ edgex-cli event list -v
+Origin               Device                 Profile                Source  Id                                    Versionable  Readings
+10 Jan 22 15:38 GMT  Random-Integer-Device  Random-Integer-Device  Int16   75f06078-e8da-4671-8938-ab12ebb2c244  {v2}         [{974a70fe-71ef-4a47-a008-c89f0e4e3bb6 1641829092129391876 Random-Integer-Device Int16 Random-Integer-Device Int16 {[] } {13342}}]
+```
+
+### `edgex-cli event count`
+Count the number of events in core data, optionally filtering by device name
+
+```
+$ edgex-cli event count -d Random-Integer-Device
+Total Random-Integer-Device events: 54
+```
+
+
+### `edgex-cli event list`
+List all events, optionally specifying a limit and offset
```

Comment:
Could we have an example or a link for limit and offset?

Reply:
Of course! It seems actually that the --limit and --offset mechanism might be broken in the CLI, so I'm going to create an issue and fix for that as well.